spec/spec.md

```diff
@@ -607,21 +646,18 @@ sign_and_send_transaction
 {
     “auth_token”: “<auth_token>”,
     “payloads”: [“<transaction>”, ...],
-    “commitment”: “<commitment_level>”,
-    "cluster": "<cluster>",
-    "skip_preflight": <skip_preflight>,
-    "preflight_commitment": "<preflight_commitment>",
+    "options": {
+        “commitment”: “<commitment_level>”,
+    }
 }
 ```
 
 where:
 
-- `auth_token`: an auth_token returned by [`authorize`](#authorize), [`reauthorize`](#reauthorize), or [`clone_authorization`](#clone_authorization) for which access to `sign_and_send_transaction` was requested
-- `transaction`: one or more base64-encoded transaction payload to sign, each with a maximum pre-encoding size of `MAX_TRANSACTION_SZ`
-- `commitment_level`: one of `processed`, `confirmed`, or `finalized`
-- `cluster`: (optional) if set, the Solana network cluster to send the signed transaction(s) to; supported values include `mainnet_beta`, `testnet`, `devnet`. If not set, the wallet endpoint determines which cluster to send transactions to.
-- `skip_preflight`: (optional) see the JSON-RPC [sendTransaction](https://docs.solana.com/developing/clients/jsonrpc-api#sendtransaction) API. If not set, defaults to `false`.
-- `preflight_commitment`: (optional) see the JSON-RPC [sendTransaction](https://docs.solana.com/developing/clients/jsonrpc-api#sendtransaction) API. If not set, defaults to `finalized`.
+- `auth_token`: an auth_token returned by [`authorize`](#authorize), [`reauthorize`](#reauthorize), or [`clone_authorization`](#clone_authorization) for which access to `sign_and_send_transactions` was requested
+- `payloads`: one or more base64-encoded transaction payload to sign
+- `options`: a JSON object, containing:
+  - `commitment`: (optional) if set, one of `processed`, `confirmed`, or `finalized`, specifying the desired commitment level that must be reached for this request to be successful. If not set, defaults to `confirmed`.
```

Comment:
So, hold on, I've been meaning to ask this. Are we actually making the wallets do transaction confirmation? This would be a new thing, for wallet adapters.

Take, for instance, the current Phantom wallet adapter for desktop web:

https://github.com/solana-labs/wallet-adapter/blob/master/packages/wallets/phantom/src/adapter.ts#L209-L210

It sends the transaction, returns the (unasserted) signature, and then leaves confirmation up to the dApp.

* Control returns to the dApp right away
* Users can perform new orthogonal transactions while waiting for the last one(s) to complete
* The dApp can care about the confirmation or not (eg. maybe it's no longer important if the user has navigated away from that screen)

Reply:
I haven't had push-back from wallets on this, but it's possible that the implications of this haven't settled in yet. Let's try and force the issue, and see what wallets say.

From a UX standpoint, I can see benefits to making the transaction confirmation experience common across dapps. However, it does differ from the other implementations of this functionality.

It's worth noting that this gives the _option_ to the dapp of having the in-wallet confirmation experience, but does not require it. They can specify `processed` here and get control back very quickly, and then perform their own wait UX, or continue on with their day, etc.

Reply:
We could always add some indication to not wait at all (a `none` value, perhaps?), but I have some wallet feedback requests out at the moment, and would like that input before we invest additional time in the details here.

Reply:
Found it. From the docs for the [`sendTransaction` RPC method](https://docs.solana.com/developing/clients/jsonrpc-api#sendtransaction):

> If the node's rpc service receives the transaction, this method immediately succeeds, without waiting for any confirmations. A successful response from this method does not guarantee the transaction is processed or confirmed by the cluster.

Reply:
Given that this PR isn't adding the `commitment` parameter, I'm going to propose that we break this sub-issue off from this PR. That'll unblock work on the agreed parts of this spec, and we can tackle the discussions with wallets on this point separately.

Reply:
Hey Phantom dev here! 👋🏽 

@sdlaver I think that's a good call. `sign_and_send_transactions` has some pretty large implications when it comes to tx simulation inside of the wallet approval notification. There is also the concept of JSON RPC batching to consider.

Generally, I'd also like to understand where this spec aligns and deviates from the standard wallet adapter interface. In an ideal world, this RPC spec is the same for any portions that use the same method. I hear there's a Discord server opening soon where I can maybe get a bit more looped in. In the meantime, I'd appreciate any links to important conversations deviations in this RPC spec vs the existing one. Thanks, y'all 🙏🏽 ❤️

Reply:
Hi @jozanza, we have https://discord.gg/solanamobile for all discussions SMS-related. There's a #wallet channel there (which reminds me, I still need to point to this PR there). Let's have a discussion on where the spec aligns, and where it differs, in that forum.

Reply:
Sounds like a plan. I'll see you there. Thanks!

Reply:
Filed https://github.com/solana-mobile/mobile-wallet-adapter/issues/134 to track closing on this discussion separately from this PR.